### PR TITLE
Add validation in DecryptTransformer constructor.

### LINKED ISSUE #15: Add validation to ensure that keyAgreementKey (and other params) are passed into the constructor
Ensure that `keyAgreementKey` is defined here:

https://github.com/digitalbazaar/minimal-cipher/blob/master/DecryptTransformer.js#L92

### PR DESCRIPTION
**Pull request by mattcollier**
Fixes #15.

### CODE REVIEW

**Review by gannan08 — APPROVED** [review]

File: DecryptTransformer.js [hunk 1]
```diff
@@ -26,6 +26,12 @@ export class DecryptTransformer {
     keyAgreement,
     keyAgreementKey
   } = {}) {
+    if(!keyAgreement) {
+      throw new SyntaxError('"keyAgreement" is a required parameter.');
+    }
+    if(!keyAgreementKey) {
+      throw new SyntaxError('"keyAgreementKey" is a required parameter.');
```
Comment (dlongley) [inline DecryptTransformer.js]:
```suggestion
      throw new TypeError('"keyAgreementKey" is a required parameter.');
```

File: DecryptTransformer.js [hunk 2]
```diff
@@ -26,6 +26,12 @@ export class DecryptTransformer {
     keyAgreement,
     keyAgreementKey
   } = {}) {
+    if(!keyAgreement) {
+      throw new SyntaxError('"keyAgreement" is a required parameter.');
```
Comment (dlongley) [inline DecryptTransformer.js]:
```suggestion
      throw new TypeError('"keyAgreement" is a required parameter.');
```

**Review by dlongley — APPROVED** [review]
Should be `TypeError`s -- then approved.

File: DecryptTransformer.js (hunk 1 above)
Comment (mattcollier) [reply DecryptTransformer.js]: I referred to this to select a `SyntaxError` for this purpose: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Missing_formal_parameter

Comment (dlongley) [reply DecryptTransformer.js]: That is not referring to missing parameters when a function is called, but rather a syntax error when declaring a function (i.e., it won't even parse as valid JS):

```js
function square(3) {
  return number * number;
};
// SyntaxError: missing formal parameter

function greet("Howdy") {
  return greeting;
};
// SyntaxError: missing formal parameter

function log({ obj: "value"}) { 
  console.log(arg)
};
// SyntaxError: missing formal parameter
```

Comment (dlongley) [reply DecryptTransformer.js]: See this instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/More_arguments_needed

**Pull request merged**